### PR TITLE
Forbid users to use empty titles for posts

### PR DESCRIPTION
**Pull request by MM263**
- Add a regex that checks if string contains anything but whitespace
- Check for whitespace-only titles on post creation and edit
- Trim whitespace from titles before saving

### CODE REVIEW

**Comment by MM263** [comment]
Also I'm not entirely sure how should I go about adding a translation. Would appreciate help on that front!

**Comment by dessalines** [comment]
Looks good to me, thanks. Could you also add this regex to the front end in this PR too? `post-form.tsx L#268`.

Its probably as simple as adding `pattern=".*\S.*"`

To make a translation: https://weblate.yerbamate.dev/

To add a term, just add one in `ui/translations/en.json`

File: server/src/api/post.rs [hunk 1]
```diff
@@ -136,6 +137,10 @@ impl Perform for Oper<CreatePost> {
       }
     }
 
+    if !is_valid_post_title(&data.name) {
+      return Err(APIError::err("invalid_post_title").into());
```
Comment (dessalines) [inline server/src/api/post.rs:141]: Add this string to `ui/translations/en.json`

**Comment by MM263** [comment]
@dessalines Added frontend validation and a translation. Unfortunately `textfield` doesn't take `pattern` prop, but I whipped up a quick util function instead

File: ui/src/components/post-form.tsx [hunk 2]
```diff
@@ -271,12 +272,19 @@ export class PostForm extends Component<PostFormProps, PostFormState> {
...
                 required
                 rows={2}
                 minLength={3}
                 maxLength={MAX_POST_TITLE_LENGTH}
               />
+              {!validTitle(this.state.postForm.name) && (
+                <div class="invalid-feedback">
+                  {i18n.t('invalid_post_title')}
+                </div>
+              )}
```
Comment (dessalines) [inline ui/src/components/post-form.tsx:287]: Nice, GJ on the bootstrap classes.

**Comment by dessalines** [comment]
Looks good! I'm testing this now.

**Comment by dessalines** [comment]
Nice job, thx.

**Pull request merged**